### DISCUSSION TITLE
Starting forever

### DISCUSSION THREAD

**Discussion by andrew222651** [redhat-developer/lsp4ij#523] [Q&A] [answered]
If the server just shows "starting" and never changes and the Traces and Logs windows are empty what does that mean?

**Comment by angelozerr** [comment]
Sometimes it means that the command which statts the language server is not correct.

I know it is not perfect and we should improve that by stopping the server and showing an error message.

Which language server you are trying to start? With which command?

**Reply by andrew222651** [reply]
Thanks for the response. I was trying to start the [erg](https://github.com/erg-lang/erg) language server with the `erg server` command.

**Reply by angelozerr** [reply]
I suggest that you open a terminal and you type your command to see if you have no errors.

**Reply by andrew222651** [reply]
No errors

**Reply by amacfie-tc** [reply]
It works in Neovim with the command `:lua =vim.lsp.start({name='erg', cmd={'erg', 'server'}})`

**Comment by angelozerr** [comment]
@amacfie if you can:

 * give me some explanation  / links how to install the erg server 
 * you[ export your command with LSP4IJ ](https://github.com/redhat-developer/lsp4ij/blob/main/docs/UserDefinedLanguageServer.md#exporting-templates) and attach the zip in  this issue 

I could try it.

**Reply by amacfie-tc** [reply]
Thanks for the response.

Installation: I use the [Nix package](https://search.nixos.org/packages?channel=24.05&show=erg&from=0&size=50&sort=relevance&type=packages&query=erg) but you can also use `cargo install erg --features els`.

The export:
[lsp4ij-export20240916.zip](https://github.com/user-attachments/files/17017676/lsp4ij-export20240916.zip)

**Reply by amacfie-tc** [reply]
([You may need Python 3 installed as well](https://github.com/erg-lang/erg?tab=readme-ov-file#requirements))

**Reply by angelozerr** [reply]
@amacfie youhave not defined a mapping, I don't understand how language server can be started?

**Reply by amacfie-tc** [reply]
There is a mapping for `*.er` files:
![image](https://github.com/user-attachments/assets/df082f3b-8d67-4d9c-ba6a-c853fab068d6)

When I open such a file, the server starts:
![image](https://github.com/user-attachments/assets/11bb25ce-c3f0-4300-b4c1-a7090a44bfb7)

**Comment by angelozerr** [comment, marked as answer]
@amacfie I have tried it but as you, no success to consume the language server.

I created an issue at https://github.com/erg-lang/erg/issues/522

**Reply by andrew222651** [reply]
Thank you!

**Reply by angelozerr** [reply]
As it is a bug from ELS, I close this dicussion.
